docs/_tooling/sphinx_extensions/sphinx_extensions/check_options.py

```diff
@@ -43,7 +43,8 @@ def check_options(
     Returns 'True' if an option is not present or a value violates its pattern
     """
 
-    required_options = get_need_type(needs_types, need["type"]).get("req_opt", None)
+    required_options = get_need_type(needs_types, need["type"]).get("req_opt", [])
+    optional_options = get_need_type(needs_types, need["type"]).get("opt_opt", [])
```

Comment:
For readability mostly:
```suggestion
    need = get_need_type(needs_types, need["type"]) # is there a better name than "need"?
    required_options: DATATYPE = need.get("req_opt", [])
    optional_options: DATATYPE = need.get("opt_opt", [])
```

Reply:
Fixed. Please resolve.